impeller/golden_tests/metal_screenshoter.mm

```diff
@@ -51,6 +51,9 @@
 
   CGImageRef cgImage = [cicontext createCGImage:flipped
                                        fromRect:[ciImage extent]];
+  if (!cgImage) {
+    return {};
```

Comment:
Instead of making MetalScreenshot exist in a world where it can be a screenshot or not I'd prefer to make it mean a screenshot, then MakeScreenshot could return a `std::optional<std::unique_ptr<MetalScreenshot>>`.

Reply:
Or just have it return nullptr.

Reply:
Why is it a unique ptr instead of shared?

Reply:
Also optional would seem to be more appropriate when the answer could validly be "no", as in there is a documentable reason not to produce a screen shot.

Failing to produce a screen shot here isn't really "the answer is no", it's more "Oops".

Reply:
> Why is it a unique ptr instead of shared?

Because there is just one owner.

> Failing to produce a screen shot here isn't really "the answer is no", it's more "Oops".

You could use fml::status_or then.  Either is fine with me.

Reply:
Why is there just one owner? Why can't a save a screen shot aside for later reference and it can be deallocated when nobody needs it any more. Note that even if the CG data needs to have one owner, MetalScreenshot could be shared.

In any case I'll close the PR. It's really only valuable while we have a resource leak that causes lots of screen shots to crash which Jason is already looking into. If it is valuable for other reasons, I'll let you manage that.